src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -840,60 +874,83 @@
         The following classes are used by the renderer when performing internal operations.
         Using the browser calculations improves rendering performance.
     */
-    // A class used by the dom-renderer.ts when testing cell for a user defined width.
+    // This class is used by the dom-renderer.ts when testing cell for a user defined width.
     .datagrid-cell-width-zero {
         // Much zero. Such !important. Wow.
         border: 0 !important;
         padding: 0 !important;
         width: 0;
         flex: 0 0 auto !important;
         min-width: 0 !important;
-    }
-
-    // This class is only applied temporarily, to improve rendering performance, while performing internal operations.
-    // Used in table-renderer.ts in noLayout with NO_LAYOUT_CLASS.
-    .datagrid-no-layout {
-        display: block;
-
-        .datagrid-body, .datagrid-row, .datagrid-cell {
-            display: none;
-        }
-
-        .datagrid-table, .datagrid-header, .datagrid-header .datagrid-row, .datagrid-column {
-            display: block;
-            flex: 1;
-        }
+        display: block!important;
     }
 
     /**
         These classes are used in table-renderer.ts when it puts the datagrid into in tableMode and computes column
         widths. NOTE: they are only applied during calculation and then removed.
      */
-    .datagrid-computing-columns-width {
+
+    /**
+     * When in calculate mode
+     * - Hide the display elements for datagrid
+     * - Switch display mode for elements with projected content to calcualte sizes.
+     */
+    .datagrid-host.datagrid-calculate-mode {
         display: block;
+        // Hide parts of the display table not used for calculation.
+        .datagrid, .datagrid-footer, .datagrid-row-master {
+            display: none;
+        }
 
-        .datagrid-body {
+        .datagrid-calculation-table {
             display: table;
             table-layout: auto;
+
+            .datagrid-calculation-header {
+                display: table-header-group;
+
+                clr-dg-column {
+                    display: table-cell;
+                    min-width: 4rem;
+                }
+
+                .datagrid-column {
```

Comment:
Are these duplicated styles? If so can we extend them instead?

Reply:
I don't thing I can extend here. I want to only override the display when `clr-dg-column` etc are in the calculation table. If I understand extension id have to do something like

```
.datagrid-calculation-header {
    display: table-header-group;
    @extend clr-dg-column; // but these styles will target the datagrid-calculation-header, not columns
}
```